.github/workflows/io-pr.yaml

```diff
@@ -0,0 +1,63 @@
+name: IO WWW PR Check
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'io/www/**'
+  pull_request:
+    types: [opened, synchronize, reopened]
+    paths:
+      - 'io/www/**'
+  workflow_dispatch:
+
+jobs:
+  tests:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: io/www
+
+    strategy:
+      matrix:
+        node-version: [22.x]
+        
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Set up Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v4
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+        cache-dependency-path: io/www/package.json
+
+    - name: Install dependencies
+      run: npm ci
+    
+    - name: Run tests
+      run: npm test
+
+  deploy:
+    needs: tests
+    if: success() && github.event_name == 'pull_request'
+    uses: ./.github/workflows/io-deploy.yaml
+    secrets:
+      workspace: ${{ secrets[format('AIO_NS_{0}', github.actor)] || secrets.AIO_NS_QA }}
```

Comment:
you will need to create 'markovicpredrag1209' workspace in io, if not already. rest of us will do the same.